app/src/main/java/com/example/pygmyhippo/organizer/EventDB.java

```diff
@@ -88,21 +92,87 @@ public void getEvent(String eventID, DBOnCompleteListener<Event> listener) {
+     * @param accountID The ID used to retrieve the matching events from the database
+     * @param listener What gets notified of results
+     * @author Kori
+     */
+    public void getOrganiserEvents(String accountID, DBOnCompleteListener<Event> listener) {
+        db.collection("Events")
+                .whereEqualTo("organiserID", accountID)
+                .get()
+                .addOnCompleteListener(task -> {
+                    if (task.isSuccessful()) {
+                        QuerySnapshot queryResult = task.getResult();
+                        ArrayList<Event> events = new ArrayList<>();
+                        queryResult.forEach(doc -> events.add(doc.toObject(Event.class)));
+                        if (events.size() == 0) {
+                            Log.d("DB", String.format("Found no events with organiser ID (%s)", accountID));
+                            listener.OnCompleteDB(events, 3, DBOnCompleteFlags.NO_DOCUMENTS.value);
+                        } else if (events.size() == 1) {
+                            Log.d("DB", String.format("Found event (%s) with organiser ID (%s)", events.get(0).getEventID(), accountID));
+                            listener.OnCompleteDB(events, 3, DBOnCompleteFlags.SINGLE_DOCUMENT.value);
+                        } else {
+                            Log.d("DB", String.format("Found %d events  with organiser ID (%s)", events.size(), accountID));
+                            listener.OnCompleteDB(events, 3, DBOnCompleteFlags.MULTIPLE_DOCUMENTS.value);
+                        }
+                    } else {
+                        Log.d("DB", String.format("Could not get event with organiser ID (%s).", accountID));
+                        listener.OnCompleteDB(new ArrayList<>(), 3, DBOnCompleteFlags.ERROR.value);
+                    }
+                });
+    }
+
+    /**
+     * This method will set the currently existing event to the new one with updated values
+     * @author Kori
+     * @param event The event we want to update
+     * @param listener The listener that initiates when the data is done updating
+     */
     public void updateEvent(Event event, DBOnCompleteListener<Event> listener) {
         db.collection("Events")
                 .document(event.getEventID())
                 .set(event)
                 .addOnCompleteListener(new OnCompleteListener<Void>() {
                     @Override
                     public void onComplete(@NonNull Task<Void> task) {
-                        ArrayList<Event> events = new ArrayList<>();
                         if (task.isSuccessful()) {
-                            events.add(event);
                             Log.d("DB", String.format("Successfully updated event with ID (%s).", event.getEventID()));
-                            listener.OnCompleteDB(events, 2, DBOnCompleteFlags.SUCCESS.value);
+                            listener.OnCompleteDB(new ArrayList<>(), 4, DBOnCompleteFlags.SUCCESS.value);
```

Comment:
in my code i use the event passed back here, but I could modify it once you make this pr to work with this implementation

Reply:
Also did you update the on complete listeners with query id 2 to query id 4 everywhere else?

Reply:
Ohhh wait no I can change that back. When I merged with main there were two updateEvents in this file so I just deleted the one I thought was mine but I guess I chose the wrong one. I'll have to fix this

Reply:
Don't worry, I will need to fix it when I merge the database stuff. Which way works better for you? Could you easily adapt your code to receive the event? or should I go ahead and change mine to remembering the event requested?

Reply:
I mean I don't think I even use the listener for anything except logging on my end. So it would be easier to go with your implementation. query ID might have to get changed tho

Reply:
It should be fixed now. Honestly I don't know when that change even came up when I was coding this 😭